### PR TITLE
BUG: fix sorting with observed=True in groupby

### LINKED ISSUE #25167: Groupby with observed=True doesn't sort
#### Code Sample, a copy-pastable example if possible

```python
>>> df = pd.DataFrame({'A': pd.Categorical(['b', 'a']), 'B': [1, 2]})
>>> df.groupby('A').sum()  # ok
   B
A
a  2
b  1
>>> df.groupby('A', observed=True).sum()  # wrong sort order
   B
A
b  1
a  2
```
#### Problem description

By default groupby has ``sort=True``, so should sort the groupings, also when using ``observed=True``. In the above example, the sorting doesn't work together with ``observed=True``.

#### Expected Output

```python
>>> df.groupby('A', observed=True).sum()
   B
A
a  2
b  1
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.7.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 78 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.24.0
pytest: 4.0.1
pip: 18.1
setuptools: 40.4.2
Cython: None
numpy: 1.15.4
scipy: None
pyarrow: None
xarray: None
IPython: 6.5.0
sphinx: None
patsy: None
dateutil: 2.7.5
pytz: 2018.7
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: 2.5.9
xlrd: 1.1.0
xlwt: None
xlsxwriter: None
lxml.etree: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by topper-123**
- [x] closes #25167
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Fix bug in #25167.

### CODE REVIEW

File: pandas/tests/groupby/test_categorical.py [hunk 1]
```diff
@@ -360,6 +360,14 @@ def test_observed(observed):
     expected = groups2.agg('mean').reset_index()
     tm.assert_frame_equal(result, expected)
 
+    # GH 25167
```
Comment (jreback) [inline pandas/tests/groupby/test_categorical.py]: make this a separate test, parameterize by sort as well

**Review by jreback — CHANGES_REQUESTED** [review]
I am not sure I buy your test example. there are actually 4 cases here observed / sort both True/False. Please enumerate what the expected results are.

File: doc/source/whatsnew/v0.24.2.rst [hunk 2]
```diff
@@ -73,7 +73,7 @@ Bug Fixes
 
 **Reshaping**
 
--
+- Fixed bug in :meth:`DataFrame.groupby` where using ``observed=True`` didn't sort properly (:issue:`25167`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.2.rst]: move to 0.25.0

**Review by jreback — CHANGES_REQUESTED** [review]
can you merge master and update to comments

**Comment by WillAyd** [comment]
Closing as stale ping if you'd like to continue

**Pull request closed without merging**